src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedCmsTests.netcoreapp.cs

```diff
@@ -0,0 +1,498 @@
+
+                Assert.True(ByteArrayContains(ret, content));
+
+                return (ret, cert);
+            }
+        }
+
+        private static (byte[] counterSignedData, X509Certificate2 firstSignerCert, X509Certificate2 counterSignerCert) GetCounterSignedData(byte[] content, Func<(X509Certificate2, AsymmetricAlgorithm)> getFirstSignerCertAndKey, Func<(X509Certificate2, AsymmetricAlgorithm)> getCounterSignerCertAndKey, Func<AsymmetricAlgorithm, bool> signedHashCalled, Func<AsymmetricAlgorithm, bool> signedHashCalledForCounterSignerKey)
+        {
+            ContentInfo contentInfo = new ContentInfo(content);
+            SignedCms cms = new SignedCms(contentInfo, detached: false);
+
+            (X509Certificate2 firstSignerCert, AsymmetricAlgorithm firstSignerKey) = getFirstSignerCertAndKey();
+            using (firstSignerKey)
+            {
+                CmsSigner firstSigner = new CmsSigner(SubjectIdentifierType.SubjectKeyIdentifier, firstSignerCert, firstSignerKey);
+                if (firstSignerKey is DSA)
+                {
+                    firstSigner.IncludeOption = X509IncludeOption.EndCertOnly;
+                    firstSigner.DigestAlgorithm = new Oid(Oids.Sha1, Oids.Sha1);
+                }
+
+                cms.ComputeSignature(firstSigner);
+                Assert.True(signedHashCalled(firstSignerKey));
+            }
+
+            byte[] signed = cms.Encode();
+            Assert.True(ByteArrayContains(signed, content));
+            SignerInfo firstSignerInfo = cms.SignerInfos[0];
+
+            (X509Certificate2 counterSignerCert, AsymmetricAlgorithm counterSignerKey) = getCounterSignerCertAndKey();
+            using (counterSignerKey)
+            {
+                CmsSigner counterSigner = new CmsSigner(SubjectIdentifierType.SubjectKeyIdentifier, counterSignerCert, counterSignerKey);
+                if (counterSignerKey is DSA)
+                {
+                    counterSigner.IncludeOption = X509IncludeOption.EndCertOnly;
+                    counterSigner.DigestAlgorithm = new Oid(Oids.Sha1, Oids.Sha1);
+                }
+
+                firstSignerInfo.ComputeCounterSignature(counterSigner);
+                Assert.True(signedHashCalledForCounterSignerKey(counterSignerKey));
+            }
+
+            byte[] counterSigned = cms.Encode();
+            Assert.True(ByteArrayContains(counterSigned, content));
+            Assert.True(counterSigned.Length > signed.Length);
+
+            return (counterSigned, firstSignerCert, counterSignerCert);
+        }
+
+        private static bool ByteArrayContains(byte[] array, byte[] subArray)
+        {
+            int lengthDiff = array.Length - subArray.Length;
+
+            if (lengthDiff < 0)
+            {
+                return false;
+            }
+
+            for (int shift = 0; shift <= lengthDiff; shift++)
+            {
+                if (subArray.SequenceEqual(array.Skip(shift).Take(subArray.Length)))
```

Comment:
I feel like this is probably an N-squared algorithm.  I'm not sure what this method is really testing, since by checking for the ContentInfo data it's really testing some other part of SignedCms.